### PR TITLE
Count team members

### PR DESCRIPTION
**Pull request by akshaymankar**
~This is not based on develop, DO NOT MERGE until it is.~
It is now.

Fixes https://github.com/zinfra/backend-issues/issues/1207

### CODE REVIEW

File: services/brig/src/Brig/Team/API.hs [hunk 1]
```diff
@@ -133,6 +135,15 @@ routes = do
...
       .&. capture "tid"
+  get "/i/teams/:tid/size" (continue teamSizeH) $
+    accept "application" "json"
+      .&. capture "tid"
+
+teamSizeH :: JSON ::: TeamId -> Handler Response
+teamSizeH (_ ::: t) = json <$> teamSize t
+
+teamSize :: TeamId -> Handler TeamSize
+teamSize t = lift $ ESIndex.teamSize t
```
Comment (fisx) [inline services/brig/src/Brig/Team/API.hs]:
```suggestion
teamSize = lift . ESIndex.teamSize
```

**Comment by fisx** [comment]
The test in brig fails:

```
  team
    size
      post /i/teams/:tid/size:                     2020-03-31T11:43:52Z, I, Known hosts: [datacenter1:rack1:127.0.0.1:9042]
2020-03-31T11:43:52Z, I, New control connection: datacenter1:rack1:127.0.0.1:9042#<socket: 3>
FAIL
        Exception: Assertions failed:
         2: Right (TeamSize 11) =/= Right (TeamSize 0)
        
        Response was:
        
        Response {responseStatus = Status {statusCode = 200, statusMessage = "OK"}, responseVersion = HTTP/1.1, responseHeaders = [("Transfer-Encoding","chunked"),("Date","Tue, 31 Mar 2020 11:43:53 GMT"),("Server","Warp/3.3.5"),("Content-Encoding","gzip"),("Content-Type","application/json")], responseBody = Just "{\"teamSize\":0}", responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}
        CallStack (from HasCallStack):
          error, called at src/Bilge/Assert.hs:72:28 in bilge-0.22.0-5aPm7hcwMbW8hs9PsPQhoJ:Bilge.Assert
          <!!, called at test/integration/API/Team.hs:82:5 in main:API.Team
```

When extracting the team id from that test and asking ES, `.hits.total == 11`, which is what we want:

```
curl -X GET "localhost:9200/_search?pretty" -H 'Content-Type: application/json' -d'
> {
>   "query": { 
>     "bool": { 
>       "must": [
>         { "match": { "team": "4452d0b9-389e-4b68-a900-c358e7806cf5" }}
>       ]
>     }
>   }
> }
> '
{
  "took" : 3,
  "timed_out" : false,
  "_shards" : {
    "total" : 17,
    "successful" : 17,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : 11,
    "max_score" : 3.0910425,
    "hits" : [
[...]
```

Which suggests it's about the acual count query, not something about insertion into ES or ES maintainenance.

File: services/brig/test/integration/API/Team.hs [hunk 2]
```diff
@@ -65,14 +65,24 @@ tests conf m b c g aws = do
...
           ],
-        testGroup
-          "sso"
+        testGroup "sso" $
           [ test m "post /i/users  - 201 internal-SSO" $ testCreateUserInternalSSO b g,
             test m "delete /i/users/:uid - 202 internal-SSO (ensure no orphan teams)" $ testDeleteUserSSO b g,
             test m "get /i/users/:uid/is-team-owner/:tid" $ testSSOIsTeamOwner b g
-          ]
+          ],
+        testGroup "size" $ [test m "post /i/teams/:tid/size" $ testTeamSize b g]
```
Comment (fisx) [inline services/brig/test/integration/API/Team.hs]:
```suggestion
        testGroup "size" $ [test m "get /i/teams/:tid/size" $ testTeamSize b g]
```

**Comment by fisx** [comment]
all good now!  @akshaymankar please take another look at what i did, then feel free to merge.

**Comment by akshaymankar** [comment]
The changes make sense, I have killed the jobs in the pipeline to make sure #964 doesn't run out of memory, so this will remain yellow (or orange) until we manually re-trigger the jobs. I will do that as soon as the other PR goes green.

**Comment by fisx** [comment]
before concourse deletes the logs:

```
bonanza                          >         svlogd:                 FAIL (0.26s)
bonanza                          >           *** Failed! Falsified (after 14 tests and 2 shrinks):
bonanza                          >           ["@2\242\168\144\154jdN\242\150\139\188\SYN","1858-11-22_00:00:00.000000000002 #l\241\149\134\165"]
bonanza                          >           [LogEvent {_logTime = Just -146138510344-07-14 16:14:21 UTC, _logOrigin = Nothing, _logTags = Tags {fromTags = fromList []}, _logMessage = Just "_00:00:00.000000000002 #l\348581"}] /= [LogEvent {_logTime = Nothing, _logOrigin = Nothing, _logTags = Tags {fromTags = fromList []}, _logMessage = Just "@2\689178jdN\615164\SYN"},LogEvent {_logTime = Just 1858-11-22 00:00:00 UTC, _logOrigin = Nothing, _logTags = Tags {fromTags = fromList []}, _logMessage = Just "#l\348581"}]
bonanza                          >           Use --quickcheck-replay=12504 --quickcheck-max-size=50 to reproduce.
```

**Comment by fisx** [comment]
Actually, I'm pretty confident the failure is not related to this PR.  Moving it [here](https://github.com/zinfra/backend-issues/issues/1266).

**Pull request merged**